src/pages/OpportunityDetails.jsx

```diff
@@ -1,14 +1,12 @@
 import React from "react";
-import { useParams } from "react-router-dom";
-import {MapPinIcon, CalendarDaysIcon, HeartIcon} from "@heroicons/react/24/outline";
-import { opportunities } from "../util/data";
+import { useParams, useLocation } from "react-router-dom";
+import { MapPinIcon, CalendarDaysIcon, HeartIcon } from "@heroicons/react/24/outline";
 
 function OpportunityDetails() {
   const { id } = useParams();
+  const location = useLocation();
 
-  const opportunityData = opportunities.find(
-    (opportunity) => opportunity.id === parseInt(id)
-  );
+  const opportunityData = location.state?.opportunityData;
```

Comment:
I wouldn't rely on state being passed in. For example, what if I just directly try to access this link with just the id? There won't be any state passed. Instead, get the id from the URL params, then call the backend API endpoint to get the data for a single event using the ID.